app/models/health/qualifying_activity.rb

```diff
@@ -541,7 +541,8 @@ def maintain_cached_values
     end
 
     def maintain_valid_unpayable
-      self.valid_unpayable = compute_valid_unpayable?
+      self.valid_unpayable_reason = compute_valid_unpayable
+      self.valid_unpayable = compute_valid_unpayable.present?
```

Comment:
why not:
`self.valid_unpayable =compute_valid_unpayable?`

Reply:
Actually, shouldn't this be:

```
self.valid_unpayable_reason = compute_valid_unpayable
self.valid_unpayable = valid_unpayable_reason.present?
```
to prevent running through the calculation twice?